.azure-pipelines.yml

```diff
@@ -3,59 +3,159 @@ trigger:
     include:
     - master
     - release-*
-  tags:
-    include:
-      - "*"
-
-jobs:
-- job: Build
-  strategy:
-    matrix:
-      linux:
-        imageName: 'ubuntu-18.04'
-        python.version: '3.x'
-        CXX: g++
-        BUILD_PYTHON_API: ON
-        BUILD_SPARK_API: ON
-      mac:
-        imageName: 'macOS-10.14'
-        python.version: '3.7'
-        CXX: clang++
-        BUILD_PYTHON_API: ON
-        BUILD_SPARK_API: ON
-        SDKROOT: '/Applications/Xcode_10.3.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk'
-
-  pool:
-    vmImage: $(imageName)
-  steps:
-  - task: UsePythonVersion@0
-    inputs:
-      versionSpec: '$(python.version)'
-      architecture: 'x64'
-  - script: printenv
-    displayName: 'Print Env'
-  - template: ci/azure-linux_mac.yml
-
-- job: Docker
-  dependsOn: Build
-  pool:
-    vmImage: 'ubuntu-18.04'
-  condition: succeeded()
-
-  strategy:
-    matrix:
-      cli:
-        dockerfile: docker/Dockerfile-cli
-        repository: tiledb/tiledbvcf-cli
-        context: .
-      python:
-        dockerfile: docker/Dockerfile-py
-        repository: tiledb/tiledbvcf-py
-        context: .
-      # dask:
-      #   dockerfile: docker/Dockerfile-dask-py
-      #   repository: tiledb/tiledbvcf-dask
-      #   context: .
-
-  steps:
-  - template: ci/build-images.yml
+    - refs/tags/*
+
+stages:
+  - stage: CI
+    condition: not(startsWith(variables['Build.SourceBranch'], 'refs/tags'))
+    jobs:
+    - job: Build
+      strategy:
+        matrix:
+          linux:
+            imageName: 'ubuntu-18.04'
+            python.version: '3.x'
+            CXX: g++
+            BUILD_PYTHON_API: ON
+            BUILD_SPARK_API: ON
+          mac:
+            imageName: 'macOS-10.14'
+            python.version: '3.7'
+            CXX: clang++
+            BUILD_PYTHON_API: ON
+            BUILD_SPARK_API: ON
+            SDKROOT: '/Applications/Xcode_10.3.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk'
+
+      pool:
+        vmImage: $(imageName)
+      steps:
+      - task: UsePythonVersion@0
+        inputs:
+          versionSpec: '$(python.version)'
+          architecture: 'x64'
+      - script: printenv
+        displayName: 'Print Env'
+      - template: ci/azure-linux_mac.yml
+
+    - job: Docker
```

Comment:
@aaronwolen I think we should move the docker to a second stage instead of a conditional job, what are your thoughts? Its not directly related to this PR but since we are moving to stages, we could adjust this one too.

Reply:
Yes, I think that makes sense too. I'm about 1/2 done with stage-based refactor.